rskj-core/src/main/java/co/rsk/peg/RepositoryBtcBlockStoreWithCache.java

```diff
@@ -108,6 +147,31 @@ public synchronized void setChainHead(StoredBlock newChainHead) {
         if (cacheBlocks != null) {
             populateCache(newChainHead);
         }
+        setMainChainBlock(newChainHead.getHeight(), newChainHead.getHeader().getHash());
```

Comment:
setChainHead seems to set the chainhead only. This is, this method is not telling you anything about forks. Probably I'm missing it, but is there a test of btc forks with depth > 1? I feel like this will not work correctly